src/logger/manager.rs

```diff
@@ -1,43 +1,52 @@
-use log::*;
-use std::io::Write;
-
 use crate::cli;
-use env_logger::fmt::{Color, Style, StyledValue};
 
-// Change style based on the message log level
-fn colored_level<'a>(style: &'a mut Style, level: Level) -> StyledValue<'a, &'static str> {
-    match level {
-        Level::Trace => style.set_color(Color::Magenta).value("TRACE"),
-        Level::Debug => style.set_color(Color::Blue).value("DEBUG"),
-        Level::Info => style.set_color(Color::Green).value("INFO "),
-        Level::Warn => style.set_color(Color::Yellow).value("WARN "),
-        Level::Error => style.set_color(Color::Red).value("ERROR"),
-    }
-}
+use tracing::*;
+use tracing_log::LogTracer;
+use tracing_subscriber::{fmt, layer::SubscriberExt, prelude::*, EnvFilter, Layer};
 
 // Start logger, should be done inside main
 pub fn init() {
-    if cli::manager::is_verbose() {
-        std::env::set_var("RUST_LOG", "debug");
-    }
+    // Redirect all logs from libs using "Log"
+    LogTracer::init_with_filter(tracing::log::LevelFilter::Trace).expect("Failed to set logger");
+
+    // Configure the console log
+    let env_filter = EnvFilter::try_from_default_env().unwrap_or_else(|_| {
+        if cli::manager::is_verbose() {
+            EnvFilter::new("debug")
```

Comment:
Can't we use `LevelFilter` here directly ?

Reply:
Unfortunately, they aren't the same thing. A filter uses levels, but isn't restricted to them, we could, for example, set `info` for our `mavlink` module, and trace for our `server` module, as `mavlink=debug,server=trace`.

Reply:
```suggestion
            EnvFilter::new(LevelFilter::DEBUG.to_string())
```
better?